coordinator/src/test/scala/filodb.coordinator/queryplanner/SingleClusterPlannerSpec.scala

```diff
@@ -309,5 +309,71 @@ class SingleClusterPlannerSpec extends FunSpec with Matchers with ScalaFutures {
     res.result.isEmpty shouldEqual true
   }
 
+  it ("should replace __name__ with _metric_ in by and without") {
 
+    val dataset = MetricsTestData.timeseriesDatasetWithMetric
+    val dsRef = dataset.ref
+    val schemas = Schemas(dataset.schema)
+
+    val engine = new SingleClusterPlanner(dsRef, schemas, mapperRef, earliestRetainedTimestampFn = 0)
+
+    val columnFilter = Seq(ColumnFilter("_metric_", Filter.Equals("http_request_duration")),
+      ColumnFilter("_ns_", Filter.Equals("myService")))
+
+    val raw = RawSeries(rangeSelector = intervalSelector, filters = columnFilter, columns = Seq("value"))
+    val windowed = PeriodicSeriesWithWindowing(raw, from, 1000, to, 5000, RangeFunctionId.Rate)
+
+    val logicalPlan1 = Aggregate(AggregationOperator.Sum, windowed, Nil, Seq("__name__"))
```

Comment:
This is ok, but see if you can parse promql instead of manually building logical plan

Reply:
Good point! Changed it